crates/router/src/connector/noon/transformers.rs

```diff
@@ -2,94 +2,185 @@ use masking::Secret;
-                    complete: item.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.request.amount,
-                    card,
-                })
-            }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
-        }
+        let payment_data = match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(req_card) => Ok(NoonPaymentData::Card(NoonCard {
+                name_on_card: req_card.card_holder_name,
+                number_plain: req_card.card_number,
+                expiry_month: req_card.card_exp_month,
+                expiry_year: req_card.card_exp_year,
+                cvv: req_card.card_cvc,
+            })),
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Payment methods".to_string(),
+            )),
+        }?;
+
+        let order = NoonOrder {
+            amount: conn_utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
+            currency: item.request.currency,
+            channel: NoonChannels::Web,
+            category: "pay".to_string(),
+            name: item.get_description()?,
+        };
+        let payment_action = match item.request.capture_method {
+            Some(enums::CaptureMethod::Manual) => NoonPaymentActions::Authorize,
+            _ => NoonPaymentActions::Sale,
+        };
+        Ok(Self {
+            api_operation: NoonApiOperations::Initiate,
+            order,
+            configuration: NoonConfiguration {
+                payment_action,
+                return_url: item.request.router_return_url.clone(),
+            },
+            payment_data,
+        })
     }
 }
 
-//TODO: Fill the struct with respective fields
 // Auth Struct
 pub struct NoonAuthType {
     pub(super) api_key: String,
+    pub(super) application_identifier: String,
+    pub(super) business_identifier: String,
 }
 
 impl TryFrom<&types::ConnectorAuthType> for NoonAuthType {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
         match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
+            types::ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => Ok(Self {
                 api_key: api_key.to_string(),
+                application_identifier: api_secret.to_string(),
+                business_identifier: key1.to_string(),
             }),
             _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
```

Comment:
need both serialize and deserialize ? what about PartialEq, do you need that also?